### ISSUE TITLE
Undoing check-ins doesn't work on Heroku

### ISSUE THREAD

**Issue by czlee** [TabbycatDebate/tabbycat#884]
I think this is a special case of preferences not being accessible from anything off the main web thread, need to investigate further.

Interestingly, it works fine on local installations, it only fails on Heroku.

Typical stacktrace;

```
2018-10-27 01:01:53,356 - daphne.server - ERROR - Exception inside application: tuple index out of range 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/sessions.py", line 179, in __call__ 
    return await self.inner(receive, self.send) 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/middleware.py", line 41, in coroutine_call 
    await inner_instance(receive, send) 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/consumer.py", line 59, in __call__ 
    [receive, self.channel_receive], self.dispatch 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/utils.py", line 52, in await_many_dispatch 
    await dispatch(result) 
  File "/app/.heroku/python/lib/python3.6/site-packages/asgiref/sync.py", line 108, in __call__ 
    return await asyncio.wait_for(future, timeout=None) 
  File "/app/.heroku/python/lib/python3.6/asyncio/tasks.py", line 339, in wait_for 
    return (yield from fut) 
  File "/app/.heroku/python/lib/python3.6/concurrent/futures/thread.py", line 56, in run 
    result = self.fn(*self.args, **self.kwargs) 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/db.py", line 13, in thread_handler 
    return super().thread_handler(loop, *args, **kwargs) 
  File "/app/.heroku/python/lib/python3.6/site-packages/asgiref/sync.py", line 123, in thread_handler 
    return self.func(*args, **kwargs) 
  File "/app/.heroku/python/lib/python3.6/site-packages/channels/consumer.py", line 105, in dispatch 
    handler(message) 
  File "/app/tabbycat/checkins/consumers.py", line 52, in broadcast_checkin 
    checkins = get_unexpired_checkins(self.tournament, window) 
  File "/app/tabbycat/checkins/utils.py", line 57, in get_unexpired_checkins 
    start = datetime.timedelta(hours=tournament.pref(window_preference_type)) 
  File "/app/tabbycat/tournaments/models.py", line 73, in pref 
    self._prefs[name] = self.preferences.get_by_name(name) 
  File "/app/.heroku/python/lib/python3.6/site-packages/dynamic_preferences/managers.py", line 54, in get_by_name 
    return self.get(self.registry.get_by_name(name).identifier()) 
  File "/app/.heroku/python/lib/python3.6/site-packages/dynamic_preferences/registries.py", line 185, in get_by_name 
    name)) 
  tuple index out of range 
```

Possibly related: https://github.com/EliotBerriot/django-dynamic-preferences/pull/157

There's a workaround in c184f4bb13081dbdb739f01be5d3a34b14d845f8, but the effect of the workaround is to hard-code the check-in window to 12 hours, so it's definitely not a fix.

**Comment by czlee** [comment]
This is fixed in eb171eb6784825fb1f52bfea394485623635b75c, and so I've also reverted c184f4bb13081dbdb739f01be5d3a34b14d845f8 in 6dbc04476ceef8c0bb0b76830df755e90a1091e7.

**Comment by czlee** [comment]
Now also in hotfix/2.2.6 (70b3c54c9e7bd5de28b6a7812e60699f0822267a). It'll obviously conflict when merged back into develop; since the change is also in develop, the resolution should be just to ignore the change to settings.py.
